Formula/millet.rb

```diff
@@ -0,0 +1,92 @@
+require "json"
+
+class Millet < Formula
+  desc "Language server for Standard ML (SML)"
+  homepage "https://github.com/azdavis/millet"
+  url "https://github.com/azdavis/millet.git",
+      tag:      "v0.9.4",
+      revision: "f99f096471887deae67e3569b30df07aa7b33acb"
+  license any_of: ["Apache-2.0", "MIT"]
+  head "https://github.com/azdavis/millet.git", branch: "main"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args(path: "crates/millet-ls")
+  end
+
+  test do
+    initialize_request = { jsonrpc: "2.0",
```

Comment:
there are many LSP examples in the repo, maybe you can just adopt the tests from the others?

Reply:
@chenrui333 I tried with one example, but this particular LSP server errors out when `stdin` is closed without the proper shutdown sequence, so it felt icky to me to write my tests around that. Wdyt?

Btw, thanks for your review 👍🏽

Reply:
@chenrui333 Also, for some reason using `pipe_output` results in a flaky test that sometimes passes and sometimes not. I think it's related to the `sync` flag on `pipe` and this server being weird.

Reply:
got you!